app/controllers/SubmitFormController.scala

```diff
@@ -38,45 +48,101 @@ class SubmitFormController @Inject()(val messagesApi: MessagesApi,
+          periodKey,
+          customerDetails.clientName,
+          sessionData.flatRateScheme,
+          VatObligation(sessionData.start, sessionData.end, sessionData.due, periodKey),
+          SubmitVatReturnForm.submitVatReturnForm.fill(sessionData),
+          isAgent = user.isAgent)
+        )
+      }
+      case _ => {
+        Ok(views.html.submit_form(
+          periodKey,
+          None,
+          sessionData.flatRateScheme,
+          VatObligation(sessionData.start, sessionData.end, sessionData.due, periodKey),
+          SubmitVatReturnForm.submitVatReturnForm.fill(sessionData),
+          isAgent = user.isAgent)
+        )
+      }
+    }
+  }
+
+  private def renderViewWithoutSessionData(periodKey: String)(implicit request: Request[_], user: User[_], hc: HeaderCarrier) = {
 
     for {
-      customerInformation <- customerInformationCall
-      obligations <- obligationsCall
+      customerInformation <- vatSubscriptionService.getCustomerDetails(user.vrn)
+      obligations <- vatObligationsService.getObligations(user.vrn)
     } yield {
       (customerInformation, obligations) match {
         case (Right(customerDetails), Right(obs)) => {
-          Ok(views.html.submit_form(
-            periodKey,
-            customerDetails.clientName,
-            customerDetails.hasFlatRateScheme,
-            obs,
-            NineBoxForm.nineBoxForm,
-            user.isAgent
-          ))
+
+          val obligationToSubmit: Seq[VatObligation] = obs.obligations.filter(_.periodKey == periodKey)
+
+          obligationToSubmit.length match {
+            case 1 => {
+              Ok(views.html.submit_form(
+                periodKey,
+                customerDetails.clientName,
+                customerDetails.hasFlatRateScheme,
+                obligationToSubmit.head,
+                SubmitVatReturnForm.submitVatReturnForm,
+                user.isAgent
+              ))
+            }
+            case _ => {
+              Logger.warn("[SubmitFormController][Show]: Length of matched obligations to period key is not equal to 1")
+              Redirect(appConfig.returnDeadlinesUrl)
+            }
+          }
         }
         case (_, _) => errorHandler.showInternalServerError
       }
     }
   }
 
-  def submit(hasFlatRateScheme: Boolean, obligation: String, periodKey: String, name: Option[String]): Action[AnyContent] =
-    (authPredicate andThen mandationStatusCheck).async { implicit user =>
-      NineBoxForm.nineBoxForm.bindFromRequest().fold(
-        failure =>
-          Future.successful(
-            Ok(
-              views.html.submit_form(periodKey, name, hasFlatRateScheme, Json.parse(obligation).as[VatObligations], failure, user.isAgent)
-            )
-          ),
-        success => Future.successful(
-          Redirect(controllers.routes.ConfirmSubmissionController.show(hasFlatRateScheme, obligation, Json.toJson(success).toString, name, periodKey))
+  def submit(periodKey: String,
```

Comment:
Please remove query parameters and retrieve again if page is invalid as discussed